src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class NullableWalker
+    {
+#if DEBUG
+        private sealed class DebugVerifier : BoundTreeWalker, IDisposable
```

Comment:
>IDisposable [](start = 62, length = 11)

It looks like instances are only created by `Verify`. Consider having a simple `Free()` method rather than a full `Dispose()` implementation.